packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -128,27 +118,30 @@ export function generateGCConfigs(
 	 * Whether GC should run or not. The following conditions have to be met to run sweep:
 	 * 1. GC should be enabled for this container.
 	 * 2. GC should not be disabled via disableGC GC option.
-	 * 3. The current GC version should be greater of equal to the GC version in the base snapshot.
-	 * These conditions can be overridden via runGCKey feature flag.
+	 * 3. The current GC version should be greater or equal to the GC version in the base snapshot.
+	 *
+	 * These conditions can be overridden via the RunGC feature flag.
 	 */
 	const shouldRunGC =
 		mc.config.getBoolean(runGCKey) ??
 		(gcEnabled && !createParams.gcOptions.disableGC && isGCVersionUpToDate);
 
 	/**
-	 * Whether sweep should run or not. The following conditions have to be met to run sweep:
+	 * Whether sweep should run or not. This refers to whether Tombstones should fail on load and whether
+	 * sweep-ready nodes should be deleted.
 	 *
-	 * 1. Overall GC or mark phase must be enabled (this.configs.shouldRunGC).
-	 * 2. Sweep timeout should be available. Without this, we wouldn't know when an object should be deleted.
-	 * 3. The driver must implement the policy limiting the age of snapshots used for loading. Otherwise
-	 * the Sweep Timeout calculation is not valid. We use the persisted value to ensure consistency over time.
-	 * 4. Sweep should be enabled for this container. This can be overridden via runSweep
-	 * feature flag.
+	 * Assuming overall GC is enabled and sweepTimeout is provided, the following conditions have to be met to run sweep:
+	 *
+	 * 1. Sweep should be enabled for this container.
+	 * 2. Sweep should be enabled for this session.
+	 *
+	 * These conditions can be overridden via the RunSweep feature flag.
 	 */
 	const shouldRunSweep =
-		shouldRunGC &&
-		sweepTimeoutMs !== undefined &&
-		(mc.config.getBoolean(runSweepKey) ?? sweepEnabled);
+		!shouldRunGC || sweepTimeoutMs === undefined
+			? false
+			: mc.config.getBoolean(runSweepKey) ??
+			  (sweepEnabled && createParams.gcOptions.enableGCSweep === true);
```

Comment:
`enableGCSweep` doesn't seem right because whether sweep is enabled for a container or not is controlled via the `gcGeneration` setting. This is telling whether sweep should run for this session or not. So, maybe `runSweep` or `shouldRunSweep`?

Reply:
We should clarify via comments and / or docs that there are 2 types of settings - one for document level and one for session level. And then the names for the settings should have that distinction in some form too.

Reply:
Yeah we need to get consistent naming.  I'm trying for consistency with the existing GC ones.  "Allow" means per-document, and there's no parallel for Sweep because of `gcGeneration`.  "Enable"/"Disable" means per-session.  Eventually `enableGCSweep` will be deprecated and replaced with `disableGCSweep` which is the parallel to `disableGC`.

I'd like to rename the properties on `IGarbageCollectorConfigs` to use the same "allow" v. "enable" language, but in a follow-up.  Maybe I'll do part of it now though, I agree that `&&` expression is confusing as-is.

Reply:
Updated the comment on enableGCSweep